mediagrains/gsf.py

```diff
@@ -151,6 +151,133 @@ def __init__(self, msg, i, major, minor):
         self.minor = minor
 
 
+class GSFBlock():
+    """Context manager for a single block in a GSF file
+
+    Decodes the tag and size of the block
+    """
+    def __init__(self, file_data):
+        self.file_data = file_data
+        self.size = None
+        self.block_start = self.file_data.tell()  # In binary mode, this should always be in bytes
+
+    def __enter__(self):
+        try:
+            self.tag = self.read_string(4)
+        except UnicodeDecodeError:
+            self.file_data.seek(-4, 1)
+            bad_bytes = self.file_data.read(4)
+            position = self.file_data.tell() - 4
+            raise GSFDecodeError(
+                "Bytes {!r} at location {} do not make a valid tag for a block".format(
+                    bad_bytes, position),
+                position)
+
+        self.size = self.read_uint(4)
+        return self
+
+    def __exit__(self, *args):
+        self.file_data.seek(self.block_start + self.size, 0)  # TODO: Real constant
+
+    def has_child_block(self, strict_blocks=True):
+        """Checks if there is space for another child block in this block
+
+        Returns true if there is space for another child block (i.e. > 8 bytes) in this block.
+        If strict_blocks=True, this block only contains other blocks rather than any other data. As a result, if there
+        are bytes left, but not enough for another block, raise a GSFDecodeError.
+        Must be used in a context manager.
+
+        :param strict_blocks: Set to True to raise if a partial block is found
+        :returns: True if there is spaces for another block
+        :raises GSFDecodeError: If there is a partial block and strict=True
+        """
+        assert self.size is not None, "in_block() only works in a context manager"
```

Comment:
in_block() -> has_child_block()

Reply:
Oops, looks like I missed a push, sorry about that